To: vim_dev@googlegroups.com
Subject: Patch 7.3.359
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.359
Problem:    Command line completion shows dict functions.
Solution:   Skip dict functions for completion. (Yasuhiro Matsumoto)
Files:	    src/eval.c


*** ../vim-7.3.358/src/eval.c	2011-10-26 13:19:23.000000000 +0200
--- src/eval.c	2011-11-30 14:53:38.000000000 +0100
***************
*** 875,881 ****
  
  #ifdef EBCDIC
      /*
!      * Sort the function table, to enable binary sort.
       */
      sortFunctions();
  #endif
--- 875,881 ----
  
  #ifdef EBCDIC
      /*
!      * Sort the function table, to enable binary search.
       */
      sortFunctions();
  #endif
***************
*** 21737,21742 ****
--- 21737,21745 ----
  	    ++hi;
  	fp = HI2UF(hi);
  
+ 	if (fp->uf_flags & FC_DICT)
+ 	    return NULL; /* don't show dict functions */
+ 
  	if (STRLEN(fp->uf_name) + 4 >= IOSIZE)
  	    return fp->uf_name;	/* prevents overflow */
  
*** ../vim-7.3.358/src/version.c	2011-11-30 14:47:12.000000000 +0100
--- src/version.c	2011-11-30 14:56:57.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     359,
  /**/

-- 
'Psychologist' -- Someone who looks at everyone else when
an attractive woman enters the room.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
